Lection1

VictorLVA opened the pull request.

VictorLVA committed:
test2

VictorLVA committed:
l1 t5

VictorLVA committed:
l1 t5

VictorLVA committed:
some changes t5

VictorLVA committed:
fix comments Calc

VictorLVA committed:
fix comments Calc

VictorLVA committed:
try to organize github

VictorLVA committed:
trying to use treemap task4 - Task4L2v2.java

VictorLVA committed:
trying to use treemap task4 - Task4L2v2.java

VictorLVA committed:
comments fixes

VictorLVA committed:
fixes

dihnatsyeu commented on Lecture2/Task1/Task1L2.java:
```diff
@@ -0,0 +1,20 @@
+        Scanner in = new Scanner ( System.in );
+        System.out.print ( "Please enter a sentence: " );
+        String sentence = in.nextLine ();
+        String reverse = new StringBuffer(sentence).reverse().toString();
```
good choice :) However i would like you to do it in another way, without using java internal mechanism.

VictorLVA replied:
fixed

dihnatsyeu commented on Lecture2/Task1/Task1L2.java:
```diff
@@ -0,0 +1,20 @@
+        System.out.println ("Lets rewrite it with uppercase: \""+sentence.toUpperCase ()+"\"");
+        if (sentence.length () < 10)
+            System.out.println ("Lets get the 5-10 symbols: \""+sentence.substring ( 4 )+"\"");
+        else System.out.println ("Lets get the 5-10 symbols: \""+sentence.substring ( 4, 10 )+"\"");
```
when you do substring, last index is exlcuded. you need to have 11 as last index

dihnatsyeu replied:
even if you have only 1 condition for if block, surround it with {} brackets to make your code more readable

VictorLVA replied:
- when you do substring, last index is exlcuded. you need to have 11 as last index - FAD
- even if you have only 1 condition for if block, surround it with {} brackets to make your code more readable - fixed

dihnatsyeu commented on Lecture2/Task2/Task2L2.java:
```diff
@@ -0,0 +1,35 @@
+            System.out.println ( MESSAGE_INVALID );
+            return;
+        }
+        if (day == 1)
```
think how you can achieve the task using array

VictorLVA replied:
fixed

dihnatsyeu commented on Lecture2/Task2/Task2L2.java:
```diff
@@ -0,0 +1,35 @@
+    private static final String MESSAGE_INVALID = "You have entered an invalid day's index";
+
+    public static void main ( String[] args ) {
+        Scanner in = new Scanner ( System.in );
```
fix formatting. You don't need spaces before and in brackets, i.e. should be:
Scanner in = new Scanner(System.in);
The same is for day = in.nextByte ( ); and below

VictorLVA replied:
fixed

dihnatsyeu commented on Lecture2/Task5/Task5L2.java:
```diff
@@ -0,0 +1,154 @@
+                System.out.println ( MESSAGE_AGAIN );
+                continue;
+            }
+            if ((Arrays.equals ( nameArray , starNameArray ) && choice.equalsIgnoreCase ( name )) || (Arrays.equals ( lastnameArray , starLastnameArray ) && choice.equalsIgnoreCase ( lastname ))) {
```
extract this long condition to readable method with boolean return type.

VictorLVA replied:
fixed

dihnatsyeu commented on Lecture2/Task5/Task5L2.java:
```diff
@@ -0,0 +1,154 @@
+                System.out.println ( MESSAGE_AGAIN );
+                continue;
+            }
+            if (choice.length ( ) != name.length ( ) && choice.length ( ) != lastname.length ( ) && choice.length ( ) != 1 || !choice.matches ( REGEX_NAME_LASTNAME )) {
```
extract this long condition to readable method with boolean return type.

VictorLVA replied:
fixed

dihnatsyeu commented on Lecture2/Task5/Task5L2.java:
```diff
@@ -0,0 +1,154 @@
+                wordProcessing ( nameArray , starNameArray , choice , name , lastnameArray , starLastnameArray , lastname , listOfWords );
+            if (choice.length ( ) == 1)
+                letterProcessing ( nameArray , starNameArray , choice , lastnameArray , starLastnameArray , listOfLetters );
+            checkResultName = Arrays.equals ( nameArray , starNameArray );
```
unnecessary validation. You can assign checkResultName/checkResultLastName in the appropriate wordMoveValidation  method.

VictorLVA replied:
FAD - this validation for the game ending
wordMoveValidation - only for word processing

but game can be ended by letter

dihnatsyeu commented on Lecture2/Task5/Task5L2.java:
```diff
@@ -0,0 +1,154 @@
+    void main ( String[] args ) throws Exception {
+        System.out.println ( "Let's start the game!!!" );
+        Console console = System.console ( );
+        String name = null;
```
Please, read task carefully. You need to guess your own name. No need to get firstName and lastName from the user's input

VictorLVA replied:
just for improvement, not interesting to play with the same words all time :)

dihnatsyeu commented on Lecture2/Task5/Task5L2.java:
```diff
@@ -0,0 +1,154 @@
+        char[] starLastnameArray = new char[ lastname.length ( ) ];
+        for ( int i = 0 ; i < starLastnameArray.length ; i++ )
+            starLastnameArray[ i ] = '*';
+        boolean checkResultName = false;
```
boolean names could be more accurate, like isFNameOpened, isLNameOpened

VictorLVA replied:
fixed

dihnatsyeu commented on Lecture2/Task5/Task5L2.java:
```diff
@@ -0,0 +1,154 @@
+
+    private static void wordProcessing ( char[] nameArray , char[] starNameArray , String choice , String name , char[] lastnameArray , char[] starLastnameArray , String lastname , List <String> listOfWords ) {
+        boolean checkRightMove = false;
+        if (!Arrays.equals ( nameArray , starNameArray ) && choice.length ( ) == name.length ( ) && choice.equalsIgnoreCase ( name ))
```
Do you really need this validation in that method:!Arrays.equals ( nameArray , starNameArray )? As I see in your code, this method is called only if words are not opened.

VictorLVA replied:
not need it
fixed

VictorLVA replied:
not needed
fixed

dihnatsyeu commented on Lecture2/Task5/Task5L2.java:
```diff
@@ -0,0 +1,154 @@
+    private static void wordProcessing ( char[] nameArray , char[] starNameArray , String choice , String name , char[] lastnameArray , char[] starLastnameArray , String lastname , List <String> listOfWords ) {
+        boolean checkRightMove = false;
+        if (!Arrays.equals ( nameArray , starNameArray ) && choice.length ( ) == name.length ( ) && choice.equalsIgnoreCase ( name ))
+            checkRightMove = wordMoveValidation ( nameArray , starNameArray );
```
not good name for variable, checkRightMove. isWordOpened -> more or less reasonable.

VictorLVA replied:
fixed

dihnatsyeu commented on Lecture2/Task5/Task5L2.java:
```diff
@@ -0,0 +1,154 @@
+        else printWrongMove ( );
+    }
+
+    private static boolean wordMoveValidation ( char[] array , char[] starArray ) {
```
no real sense in this method.  If this method did actual array copying, then yes - method is reasonable

VictorLVA replied:
method name fixed

dihnatsyeu commented on Lecture2/Task5/Task5L2.java:
```diff
@@ -0,0 +1,154 @@
+        System.out.println ( "Let's start the game!!!" );
+        Console console = System.console ( );
+        String name = null;
+        while (name == null)
```
you can reduce number of parameters that are required to be sent to the methods by assigning it to the class.

VictorLVA replied:
fixed

VictorLVA committed:
Fixes:
- string reverse without using java internal mechanism;
- even if only 1 condition for block, surround it with {} brackets;
- print day by index (using array)
- fix formatting, no spaces before and in brackets

VictorLVA committed:
Fixes:
- not good name for variable, checkRightMove. isWordOpened -> more or less reasonable;
- boolean names could be more accurate, like isFNameOpened, isLNameOpened

VictorLVA committed:
Fixes:
- no real sense in this method. If this method did actual array copying, then yes - method is reasonable - method name fixed

VictorLVA committed:
Fixes:
- extract this long condition to readable method with boolean return type.

VictorLVA committed:
comments fixes
